tests/unit/core/components/TestAMInstallButton.js

```diff
@@ -393,19 +393,48 @@ describe(__filename, () => {
     const enable = sinon.spy();
     const install = sinon.spy();
 
-    const root = render({ addon, enable, install });
+    const root = render({
+      addon,
+      enable,
+      install,
+      getAddonEnabledStatus: sinon.stub().returns(false),
+    });
 
     const event = createFakeEvent();
 
     const installButton = root.find('.AMInstallButton-button');
     await installButton.simulate('click', event);
 
-    sinon.assert.calledOnce(install);
+    await sinon.assert.calledOnce(install);
```

Comment:
oh, we really have to `await` here?

Reply:
yeh it wasn't hitting enable without this..should I be doing something else?

Reply:
I see. I think we could avoid the `await` here and above with `setImmediate()`.

Reply:
The problem I see here is that the test case above does not `await` the `install` and therefore no matter what happens the assertion `sinon.assert.notCalled(enable);` will always be correct. That means a bug will never be caught.

Reply:
@kumar303 would you have an idea to make it better?

Reply:
> I see. I think we could avoid the `await` here and above with `setImmediate()`.

IMO `setImmediate()` is a hack and we shouldn't invest code in it.